### DISCUSSION TITLE
AWS service-catalog-hub product version Datasource

### DISCUSSION THREAD

**Discussion by sjvaiz** [renovatebot/renovate#33192] [Suggest an Idea]
### Tell us more.

I would like to propose a new datasource, similar to [aws-rds](https://docs.renovatebot.com/modules/datasource/aws-rds/), that is able to query AWS's service-catalog api's [DescribeProductAsAdmin](https://docs.aws.amazon.com/servicecatalog/latest/dg/API_DescribeProductAsAdmin.html) for retrieving information about specified product versions. 

We have several in-house service-catalog-products that get version updates and we would like renovate to able to be to track the releases.

Example usage we have is as follows:

```terraform
resource "aws_servicecatalog_provisioned_product" "my_installation" {
  name                       = "MyInstallation"
  product_name               = "Product Name"
  provisioning_artifact_name = "v1" #has to be updated by renovate
}
```

**Comment by rarkins** [comment]
I have the same question as here: https://github.com/renovatebot/renovate/discussions/33095#discussioncomment-11577260

I would like to avoid an exponential increase in AWS-related datasources. Are these definitely not able to combine into a single datasource?

**Comment by sjvaiz** [comment]
In order to combine AWS-related datasources, one suitable approach imo is #23410. However, in that case, only self-hosted instances can benefit from it. 
Moreover, each AWS api requires its own API client instance which limits us to combine multiple datasources into one.
I would also like to point out that `service catalog products` is de facto when it comes to shipping resources in AWS and warrants a customDatasource.
